rpm/src/test/java/org/eclipse/packager/rpm/signature/RpmFileSignatureProcessorTest.java

```diff
@@ -0,0 +1,113 @@
+
+        RpmFileSignatureProcessor signatureProcessor = new RpmFileSignatureProcessor();
+        ByteArrayOutputStream signedPackage = signatureProcessor.perform(rpmStream, privateKeyStream, passPhrase);
+        byte[] bytes = signedPackage.toByteArray();
+
+        File resultDirectory = new File(RESULT_DIR);
+        resultDirectory.mkdir();
+        File signedRpm = new File(RESULT_FILE_PATH);
+        signedRpm.createNewFile();
+        FileOutputStream resultOut = new FileOutputStream(signedRpm);
+        resultOut.write(bytes);
+        resultOut.close();
+
+        RpmInputStream initialRpm = new RpmInputStream(new FileInputStream(rpm));
+        initialRpm.available();
+        initialRpm.close();
+        InputHeader<RpmSignatureTag> initialHeader = initialRpm.getSignatureHeader();
+        RpmInputStream rpmSigned = new RpmInputStream(new ByteArrayInputStream(bytes));
+        rpmSigned.available();
+        rpmSigned.close();
+        InputHeader<RpmSignatureTag> signedHeader = rpmSigned.getSignatureHeader();
+
+        int initialSize = (int) initialHeader.getEntry(RpmSignatureTag.SIZE).get().getValue();
+        int initialPayloadSize = (int) initialHeader.getEntry(RpmSignatureTag.PAYLOAD_SIZE).get().getValue();
+        String initialSha1 = initialHeader.getEntry(RpmSignatureTag.SHA1HEADER).get().getValue().toString();
+        String initialMd5 = Rpms.dumpValue(initialHeader.getEntry(RpmSignatureTag.MD5).get().getValue());
+
+        int signedSize = (int) signedHeader.getEntry(RpmSignatureTag.SIZE).get().getValue();
+        int signedPayloadSize = (int) signedHeader.getEntry(RpmSignatureTag.PAYLOAD_SIZE).get().getValue();
+        String signedSha1 = signedHeader.getEntry(RpmSignatureTag.SHA1HEADER).get().getValue().toString();
+        String signedMd5 = Rpms.dumpValue(signedHeader.getEntry(RpmSignatureTag.MD5).get().getValue());
+        String pgpSignature = Rpms.dumpValue(signedHeader.getEntry(RpmSignatureTag.PGP).get().getValue());
+
+        assertEquals(initialSize, signedSize);
+        assertEquals(initialPayloadSize, signedPayloadSize);
+        assertEquals(initialSha1, signedSha1);
+        assertEquals(initialMd5, signedMd5);
+        assertNotNull(pgpSignature);
+    }
+
+    @Test
+    @Order(2)
+    public void verify_rpm_signature() throws IOException, PGPException {
+        File public_key = new File("src/test/resources/key/public_key.txt");
+        File signedRpm = new File(RESULT_FILE_PATH);
+        if (!public_key.exists() || !signedRpm.exists()) {
+            fail("Input files signedRpm or public_key does not exist");
+        }
+        InputStream publicKeyStream = new FileInputStream(public_key);
+        ArmoredInputStream armoredInputStream = new ArmoredInputStream(publicKeyStream);
+        PGPPublicKeyRing publicKeyRing = new BcPGPPublicKeyRing(armoredInputStream);
+        PGPPublicKey publicKey = publicKeyRing.getPublicKey();
+        // TODO Signature Check
```

Comment:
It's ok if the test isn't there yet. But I think in this case, it should be removed.

Reply:
It is to keep the test file logic. Read rpm -> sign it -> write signed rpm -> read it -> verify signature -> remove it. If I remove this test, write the rpm after signing it became useless.

Reply:
Or complete the test before accepting the PR...